MantidQt/CustomInterfaces/src/Reflectometry/ReflSaveTabPresenter.cpp

```diff
@@ -0,0 +1,181 @@
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTabPresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflSaveTabView.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Run.h"
+#include "Poco/File.h"
+#include "Poco/Path.h"
+
+#include <regex>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+using namespace Mantid::API;
+
+Mantid::Kernel::Logger ReflSaveTabPresenter::g_log("ReflSaveTabPresenter");
+
+/** Constructor
+* @param view :: The view we are handling
+*/
+ReflSaveTabPresenter::ReflSaveTabPresenter(IReflSaveTabView *view)
+    : m_view(view) {
+
+  FrameworkManager::Instance();
+  saveAlgs = {"SaveReflCustomAscii", "SaveReflThreeColumnAscii",
+              "SaveANSTOAscii", "SaveILLCosmosAscii"};
+  saveExts = {".dat", ".dat", ".txt", ".mft"};
+}
+
+/** Destructor
+*/
+ReflSaveTabPresenter::~ReflSaveTabPresenter() {}
+
+void ReflSaveTabPresenter::notify(IReflSaveTabPresenter::Flag flag) {
+  switch (flag) {
+  case populateWorkspaceListFlag:
+    populateWorkspaceList();
+    suggestSaveDir();
```

Comment:
Testing functionality I can see that if I have something like this:

![savedir](https://cloud.githubusercontent.com/assets/8956985/20343952/8cd05ab2-abe8-11e6-9aa8-333b8fb7bb8f.png)

If I now click on `Refresh` the save path is cleared.